NBitcoin/Protocol/NodesCollection.cs

```diff
@@ -64,6 +64,25 @@ public void PushMessage(IncomingMessage message)
 
 			#endregion
 		}
+
+		/// <summary>
+		/// Provides a comparer to specify how nodes are compared for equality
+		/// </summary>
+		public class NodeComparer:IEqualityComparer<Node>
+		{
+			public bool Equals(Node a, Node b)
+			{
+				bool same = a.RemoteSocketAddress.MapToIPv6().ToString() == b.RemoteSocketAddress.MapToIPv6().ToString() && a.RemoteSocketPort == b.RemoteSocketPort;
+				return same;
+			}
+
+			public int GetHashCode(Node x)
+			{
+				int hash = x.RemoteSocketPort.GetHashCode() ^ x.RemoteSocketAddress.MapToIPv6().ToString().GetHashCode();
```

Comment:
Same here in checking that x is not null.
Maybe it's nicer if it's called `node` rather than `x`.

Reply:
Done